tests/framework/extensions/clusters/clusters.go

```diff
@@ -74,6 +75,19 @@ func IsProvisioningClusterReady(event watch.Event) (ready bool, err error) {
 	return ready && updated, nil
 }
 
+// IsProvisioningClusterReadyAndUpdated is basic check function that would be used for checking if the cluster is ready after an update.
+func IsProvisioningClusterReadyWithPoll(cluster *apisV1.Cluster) (ready bool, err error) {
```

Comment:
Rishabh, as Israel mentioned. I think using the `IsProvisioningClusterReady` with the watch interface and removing this would be better since the watched properties are identical.

Reply:
Agreed

Reply:
Removed the IsProvisioningClusterReadyWithPoll function.